pkg/ingester/retention.go

```diff
@@ -289,12 +289,19 @@ func (dc *diskCleaner) CleanupBlocksWhenHighDiskUtilization(ctx context.Context)
 }
 
 // isBlockDeletable returns true if this block can be deleted.
-func (dc *diskCleaner) isBlockDeletable(block *tenantBlock) bool {
+func (dc *diskCleaner) isBlockDeletable(block *tenantBlock, deleteOnlyUploaded bool) bool {
```

Comment:
I see that `deleteOnlyUploaded` is always `true`. The only exception is the test `TestDiskCleaner_isBlockDeletable`. I think we should remove the argument and update the test

Reply:
I have updated some code, `deleteOnlyUploaded` helps function to detect if function should ignore upload or not. In terms of function usage in  `DeleteUploadedBlocks` should delete only uploaded blocks and for `CleanupBlocksWhenHighDiskUtilization` should delete any block(Prefarably uploaded, which is handled by sorting blocks), This has been updated in latest commits. In terms of test, some test are failing, I am investigating on that